libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/security/Encryptor.java

```diff
@@ -92,53 +94,91 @@ public static Cipher getDecryptingCipher(String encryptionKey, byte[] iv)
     }
 
     private static Cipher getDecryptingCipher(byte[] keyBytes, byte[] iv)
-        throws InvalidAlgorithmParameterException, InvalidKeyException {
-        final Cipher cipher = getBestCipher();
+            throws InvalidAlgorithmParameterException, InvalidKeyException {
+        final Cipher cipher = getBestCipher(AES_GCM_CIPHER);
+        final SecretKeySpec skeySpec = new SecretKeySpec(keyBytes, cipher.getAlgorithm());
+        final IvParameterSpec ivSpec = new IvParameterSpec(iv);
+        cipher.init(Cipher.DECRYPT_MODE, skeySpec, ivSpec);
```

Comment:
Don't you need a GCMParameterSpec ? See https://gist.github.com/praseodym/f2499b3e14d872fe5b4a

Reply:
Yes, I haven't gotten there yet.

Reply:
We need to use `SecretKeySpec` because we have raw bytes for the key. We don't need `GcmParameterSpec` as long as we update `AAD` in the `Cipher` directly.